python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1283,6 +1283,22 @@ def handle_run_event(self, run_id: str, event: "DagsterEvent"):
     def add_run_tags(self, run_id: str, new_tags: Dict[str, str]):
         return self._run_storage.add_run_tags(run_id, new_tags)
 
+    @traced
+    def add_asset_event_tags(self, asset_event_id: int, new_tags: Dict[str, str]):
```

Comment:
My current PRs only support adding tags when writing the initial event into storage. Just as a heads up, you'll need to make changes to allow updating tags for existing events after I merge